pkg/runner-provisioner/provisioner.go

```diff
@@ -95,7 +96,7 @@ func (p *RunnerProvisioner) getRealVMIP(vmIP string) (string, error) {
 
 func (p *RunnerProvisioner) deleteVM(ctx context.Context, runnerName string) {
 	p.logger.Infof("deleting Orka VM with name %s", runnerName)
-	err := p.orkaClient.DeleteVM(ctx, runnerName)
+	err := backoff.Retry(func() error { return p.orkaClient.DeleteVM(ctx, runnerName) }, backoff.NewExponentialBackOff())
```

Comment:
One thing we could improve here is not to retry when the VM does not exist. So in cases that say "does not exist" we can skip the retry